src/OnlineSales/Controllers/BaseController.cs

```diff
@@ -110,7 +110,7 @@ public virtual async Task<ActionResult> Delete(int id)
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
-        public virtual async Task<ActionResult<List<TD>>> Get([FromQuery] IDictionary<string, string>? parameters)
+        public virtual async Task<ActionResult<List<TD>>> Get([FromQuery] string? query)
         {
             int limit = apiSettingsConfig.Value.MaxListSize;
             var queryCommands = this.Request.QueryString.HasValue ? this.Request.QueryString.ToString().Substring(1).Split('&').Select(s => HttpUtility.UrlDecode(s)).ToArray() : new string[0];

```

Comment:
it is better to move query parsing to separate method, so it can be used in other other actions of controllers inherited BaseController